app/src/main/java/de/tum/in/tumcampusapp/activities/RoomFinderDetailsActivity.java

```diff
@@ -218,24 +191,125 @@ protected void onLoadFinished(Optional<File> result) {
         getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, mImage).commit();
 
         if (getSupportActionBar() != null) {
-            getSupportActionBar().setTitle(roomInfo.getString(TUMRoomFinderRequest.KEY_ROOM_TITLE));
-            getSupportActionBar().setSubtitle(roomInfo.getString(TUMRoomFinderRequest.KEY_BUILDING_TITLE));
+            getSupportActionBar().setTitle(room.getInfo());
+            getSupportActionBar().setSubtitle(room.getAddress());
         }
+
         showLoadingEnded();
-        new Thread(new Runnable() {
+
+        startLoadingMapList(room.getArch_id());
+    }
+
+    private void onMapListLoadFinished(Optional<List<RoomFinderMap>> result){
+        if(!result.isPresent()){
+            if (NetUtils.isConnected(this)) {
+                showErrorLayout();
+            } else {
+                showNoInternetLayout();
+            }
+            return;
+        }
+
+        mapsList = result.get();
+        if (mapsList.size() > 1) {
+            mapsLoaded = true;
+        }
+
+        supportInvalidateOptionsMenu();
+    }
+
+    private Optional<List<RoomFinderMap>> onLoadMapListInBackground(String archId){
+        try {
+            Optional<List<RoomFinderMap>> data =
+                    Optional.of(TUMCabeClient.getInstance(this).fetchAvailableMaps(archId));
+            if(data.isPresent()){
+                return data;
+            }
+        } catch (IOException | NullPointerException e) {
+            Utils.log(e);
+        }
+
+        return Optional.absent();
+    }
+
+    final void startLoadingMapList(String... params){
+        if (mapListAsyncTask != null) {
+            mapListAsyncTask.cancel(true);
+        }
+
+        mapListAsyncTask = new AsyncTask<String, Void, Optional<List<RoomFinderMap>>>() {
```

Comment:
You could just use Retrofit Callbacks / Asynchronous Requests for this - no?

https://futurestud.io/tutorials/retrofit-synchronous-and-asynchronous-requests